### PR TITLE
Sessions.getByBearerToken(): check format before hitting db

### LINKED ISSUE #1262: Validate token format before using them
* sessions
* field keys
* enketo access tokens
* CSRF tokens

Related: https://github.com/getodk/central-backend/pull/1234

### PR DESCRIPTION
**Pull request by alxndrsn**
Closes #1262

### CODE REVIEW

**Comment by matthew-white** [comment]
This looks like a reasonable change to me. :+1:

I see that the PR is marked as a draft. Is there work that's still left to do for this PR?

File: lib/util/crypto.js [hunk 1]
```diff
@@ -43,6 +43,8 @@ const generateToken = (bytes = 48) => randomBytes(bytes).toString('base64')
   .replace(/\//g, '!')
   .replace(/\+/g, '$');
 
+const isValidStandardLengthToken = (token) => /^[A-Za-z0-9!$]{64}$/.test(token);
```
Comment (matthew-white) [inline lib/util/crypto.js:46]: What do you think about a name that's a little shorter? How about `isValidToken` or `matchesTokenFormat`?

Comment (alxndrsn) [reply lib/util/crypto.js:46]: I think `isValidToken()` would be fine, except that this is a valid token:

```js
require('./lib/util/crypto').generateToken(0)
''
```

In the current codebase, there are zero uses of `generateToken()` with an argument, so to allow `isValidToken` as the name for this function, I think step 1 would be to remove the `bytes` argument from `generateToken()`.  Would that be an acceptable change?

Comment (alxndrsn) [reply lib/util/crypto.js:46]: PR to remove `bytes` arg at git push --set-upstream origin generate-token-no-args

Comment (matthew-white) [reply lib/util/crypto.js:46]: Just approved #1264!

**Review by matthew-white — APPROVED** [review]

**Pull request merged**